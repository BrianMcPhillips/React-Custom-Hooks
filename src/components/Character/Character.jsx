import React from 'react';
import PropTypes from 'prop-types';

const Character = () => (
  <h1>hello world</h1>
);

Character.propTypes = {
  name: PropTypes.string.isRequired
};

export default Character;
