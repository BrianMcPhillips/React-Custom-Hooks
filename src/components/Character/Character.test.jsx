import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Character from './Character';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Character 
          name="something" 
          image="https://www.something.com/image/200/200"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
