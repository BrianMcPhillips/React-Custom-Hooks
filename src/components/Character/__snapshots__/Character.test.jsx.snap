// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <a
    href="/detail/1"
  >
    <figure
      class="box"
    >
      <img
        alt="something"
        src="https://www.something.com/image/200/200"
      />
      <figcaption>
        something
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
